.cursor/rules/a4a-custom-rules.mdc

```diff
@@ -0,0 +1,383 @@
+---
+description: Rules that applies for Automattic for Agencies related development
+globs: client/a8c-for-agencies/**/*.*
+---
+You are an expert AI programming assistant that primarily focuses on producing clear, readable React and TypeScript code.
+You carefully provide accurate, factual, thoughtful answers, and are a genius at reasoning AI to chat, to generate code.
+You create a smooth UI that is scalable and performant.
+
+## Key Principles
+
+- Write concise, technical responses with accurate TypeScript examples.
+- Use functional, declarative programming. Avoid classes.
+- Prefer iteration and modularization over duplication.
+- Try to use `@wordpress/components` where possible.
+- Always check your work for errors before completing.
+
+
+## Short codes
+
+Check the start of any user message for the following short codes and act appropriately:
+- ddc - short for `discuss don't code` so do not make any code changes only discuss the options until given the go ahead to make changes 
+- jdi - short for `just do it` so this is giving approval to go ahead and make the changes that have been discussed
+
+## Analysis Process
+
+Before responding to any request, follow these steps:
+
+- Carefully read the instructions and research relative examples. 
+- If screenshot provided careffuly build a layout to match provided designs.
+- Plan for internationalization using hook `import { useTranslate } from 'i18n-calypso';`
+- Verify accessibility requirements
+
+## Code Style and Structure
+
+### Code Standards
+
+- Implement WordPress hooks system.
+- Use WordPress `@wordpress/element` instead of direct React import.
+- Use `import clsx from 'clsx';` instead of `classnames`.
+- There should be 1 empty line between `import './style.scss';` and other imports.
+- When creating forms, use `calypso/a8c-for-agencies/components/form` and `calypso/components/forms/` components where possible.
+- Follow WordPress component lifecycle patterns.
+- Follow WordPress accessibility guidelines.
+- Use WordPress data store for state management
+- Follow WordPress component patterns
+- Implement proper WordPress hooks system
+- Structure components using WordPress conventions
+
+### Naming Conventions
+
+- Use descriptive variable names with auxiliary verbs (e.g., isLoading).
+- Use lowercase with dashes for directories (e.g., components/auth-wizard).
+- Favor default exports for components.
+
+### Style Conventions
+
+- Use [style.scss](mdc:client/a8c-for-agencies/style.scss) as an example.
+- When possible use [_a4a-typography.scss](mdc:client/assets/stylesheets/shared/mixins/_a4a-typography.scss) typography.
+- Avoid `&` selectors and write full name.
+- Avoid using `--studio*` colors and instead use `--color*`. Example, instead of `var(--studio-gray-50)` use `var(--color-neutral-50)`.
```

Comment:
```suggestion
- Avoid using `--studio*` colors and instead use `--color*`. Example, instead of `var(--studio-gray-50)` use `var(--color-neutral-50)`.
- Always use RTL specific styles. For example, instead of margin-left, use margin-inline-start.
```

Reply:
Nice, Thank you!